crypto/s2n_certificate.c

```diff
@@ -383,24 +383,41 @@ int s2n_cert_chain_and_key_free(struct s2n_cert_chain_and_key *cert_and_key)
     return 0;
 }
 
-int s2n_send_cert_chain(struct s2n_stuffer *out, struct s2n_cert_chain *chain, uint8_t actual_protocol_version)
+int s2n_send_cert_chain(struct s2n_connection *conn, struct s2n_stuffer *out, struct s2n_cert_chain_and_key *chain_and_key)
 {
     notnull_check(out);
+    notnull_check(chain_and_key);
+    struct s2n_cert_chain *chain = chain_and_key->cert_chain;
     notnull_check(chain);
-    if (actual_protocol_version == S2N_TLS13) {
-        GUARD(s2n_stuffer_write_uint24(out, chain->chain_size + s2n_certificate_extensions_size(chain->head)));
-    }
-    else {
-        GUARD(s2n_stuffer_write_uint24(out, chain->chain_size));
+    struct s2n_cert *cur_cert = chain->head;
+    notnull_check(cur_cert);
+    uint32_t cert_chain_size = chain->chain_size;
+
+    /* Certificate extensions are enabled for TLS 1.3 */
+    uint8_t actual_protocol_version = conn->actual_protocol_version;
```

Comment:
nit: this variable is only used once, we could avoid it by just comparing conn->actual_protocol_version.